### PR TITLE
Implement new nav for docs website + Cody revamp

### PR DESCRIPTION
**Pull request by MaedahBatool**
This PR ships:

1. New structure/layout for our root left nav of our docs website ([preview](https://docs.sourcegraph.com/@new-nav))
2. Revamped IA (information architecture) for Cody docs ([preview](https://docs.sourcegraph.com/@new-nav/cody/overview))

- For (1), it implements a better-segregated view of our products with an Overview page for every section to kick off things. This will change in the long run. This is iterative, and I aim to improve and trim it as we go.

- The (2nd) one is aimed towards a greater goal while keeping in mind Cody's docs. This is the proposed V1 iteration of the revamped nav. It's based on a user journey of a first-timer developer (persona) who lands on our Cody docs. This nav makes sure the developer gets a well-guided product tour with the right amount of info at the right place. 

## Preview

- [Here](http://docs.sourcegraph.com/@new-nav)

## Test plan

Docs things

### CODE REVIEW

File: doc/sidebar.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ Keep it as a single list with at most 2 levels. (Anything else may not render co
 - [Overview](cody/overview/index.md)
   - [Cody for VSCode](cody/overview/install-vscode.md)
```
Comment (jdorfman) [inline doc/sidebar.md:30]: VS Code

**Review by taylorsperry — APPROVED** [review]
Love the simplification here! I know we have more content-specific updates to make (and I'm excited to work with you on that 😄) but for now the re-architecting is a huge improvement. Thank you!

File: doc/cody/overview/install-jetbrains.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Installing Cody in Jetbrains (Experimental)
+# Install Cody for Jetbrains <span class="badge badge-experimental" style="margin-left: 0.5rem; vertical-align:middle;">Experimental</span>
```
Comment (vdavid) [inline doc/cody/overview/install-jetbrains.md:1]:
```suggestion
# Install Cody for JetBrains <span class="badge badge-experimental" style="margin-left: 0.5rem; vertical-align:middle;">Experimental</span>
```

File: doc/cody/explanations/index.md [hunk 3]
```diff
@@ -1,14 +1,14 @@
 # Explanations
 
 - Installing or turning on Cody
-  - [Enabling Cody for Sourcegraph Enterprise](enabling_cody_enterprise.md)
+  - [Enabling Cody for Sourcegraph Enterprise](./../overview/enable-cody-enterprise.md)
```
Comment (jdorfman) [inline doc/cody/explanations/index.md:4]: try removing `./` to fix the Buildkite errors

Comment (MaedahBatool) [reply doc/cody/explanations/index.md:4]: I don't think this is the issue. It's something exit status 3.

**Comment by jhchabran** [comment]
Porting here the essence of our slack exchange: the new UI redirects with status 308, not 301, but the docsite link checker only accepts 301 as a valid thing. We could patch the docsite checker, but that might delay merging this PR, which IIRC can easily end up creating conflicts and being a pain to update.

**Pull request merged**